src/coreclr/gc/gc.cpp

```diff
@@ -24211,6 +24211,9 @@ size_t gc_heap::get_promoted_bytes()
 
     dprintf (3, ("h%d getting surv", heap_number));
     size_t promoted = 0;
+#ifdef _MSC_VER
+#pragma loop(no_vector)
```

Comment:
Add a link to the MSVC bug that this is working around?

Reply:
Yes, I think I will add it to the description and merge as soon as the tests finish so that this stops impacting everyone, and then I can add a comment when the timing doesn't matter (and hopefully we'll know more -- latest toolchains don't repro this anymore, but we'd like to know if it was directly fixed.  Andy filed https://developercommunity.visualstudio.com/t/Bad-codegen-in-coreclr-GC-code/10625094)

Reply:
https://devdiv.visualstudio.com/DevDiv/_git/msvc/commit/e72c28af106a0efbc93eb67b7b21c433e8749424

I haven't checked yet.  We might be able to remove the workaround already.

Reply:
We've stopped using preview compilers.  The codegen bug was fixed.  This workaround was removed in #101995.